src/utils/confirmBeforeExit.ts

```diff
@@ -0,0 +1,46 @@
+import { Dialog } from 'quasar';
+import { onBeforeRouteLeave } from 'vue-router';
+
+type ConfirmBeforeExit = {
+  title?: string;
+  message?: string;
+  currentRouteParent: string;
+};
+
+export const confirmBeforeExit = ({
+  title = '¿Estás seguro de que quieres abandonar esta página?',
+  message = 'Si sales de la página, se perderán los cambios',
+  currentRouteParent,
+}: ConfirmBeforeExit) => {
+  onBeforeRouteLeave((to, from, next) => {
+    const routeParent = to.matched[0];
```

Comment:
yes, matched will always have an element, since matched are the route levels sorted from parent to child order, for example:

`/foo/bar/child`

matched = [ '/foo', '/bar', '/child' ] 

https://v3.router.vuejs.org/api/#the-route-object